core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -184,6 +184,34 @@ class GroupMetadataManager(brokerId: Int,
     }
   }
 
+  /**
+    * Delete groups
+    */
+  def deleteGroups(groupIds: Seq[String]): Map[String, Errors] = {
+    var eligibleGroups: Seq[GroupMetadata] = Seq()
+    var result: Map[String, Errors] = Map()
+
+    groupIds.foreach { groupId =>
+      if (!groupMetadataCache.contains(groupId))
+        result += (groupId -> Errors.GROUP_ID_NOT_FOUND)
+      else {
+        val group = groupMetadataCache.get(groupId)
+        if (!group.is(Empty))
+          result += (groupId -> Errors.NON_EMPTY_GROUP)
+        else
+          eligibleGroups ++= Seq(group)
+      }
+    }
+
+    if (eligibleGroups.nonEmpty) {
+      cleanupGroupMetadata(None, eligibleGroups, Long.MaxValue)
```

Comment:
I don't think passing `None` works. Looking at `cleanupGroupMetadata`, that would just result in removal of the expired offsets.